script/helperFunctions.sh

```diff
@@ -2057,8 +2088,8 @@ function getIncludedAndSortedFacetContractsArray() {
   # get all facet contracts
   FACET_CONTRACTS=($(getIncludedFacetContractsArray "$EXCLUDE_CONFIG"))
 
-  # convert CORE_FACETS into an array
-  CORE_FACETS_ARRAY=($(echo "$CORE_FACETS" | tr ',' ' '))
+  # Get core facets from global.json
+  CORE_FACETS_ARRAY=($(getCoreFacetsArray))
```

Comment:
did you try to run this with an empty core facets array?
If I am not mistaken, we need to check the return code for the getCoreFacetsArray function.
Otherwise it might be possible that we call this helper function and it fails (but will not end the script due to the way bash scripts behave when a subshell fails). And then the subsequent code will run into unexpected behaviour.

I might be wrong but please could you check?

Reply:
thanks for pointing that out. I actually removed the empty array check since CORE_FACETS could also have no elements before but I getting to the point I didn't realize that the script would keep running if the subfunction fails, you are right, I didn't know about that.  I've now updated things to use the `checkFailure` function to catch any failure from `getCoreFacetsArray`. I also made a few changes to `getCoreFacetsArray`